hphp/runtime/vm/jit/vasm-simplify.cpp

```diff
@@ -524,6 +524,24 @@ bool simplify(Env& env, const pop& inst, Vlabel b, size_t i) {
 
 ///////////////////////////////////////////////////////////////////////////////
 
+bool simplify(Env& env, const movzbl& inst, Vlabel b, size_t i) {
+  // movzbl{s, d}; shrli{2, s, d} --> ubfmli{2, 7, s, d}
+  return if_inst<Vinstr::shrli>(env, b, i + 1, [&](const shrli& sh) {
+    if (!(arch() == Arch::ARM &&
+      sh.s0.l() == 2 &&
+      env.use_counts[inst.d] == 1 &&
+      inst.d == sh.s1)) return false;
+
+  return simplify_impl(env, b, i, [&] (Vout& v) {
+    v << copy{inst.s, inst.d};
+    v << ubfmli{2, 7, inst.d, sh.d, sh.sf, sh.fl};
```

Comment:
@mxw - That's a good question.  I'm not seeing anywhere that the flag is being set/used
by either of the shrli/shrqi instructions.  The underlying ARM instruction doesn't touch any
flags.  I'll check further, but it's probably safe to remove from ubfmli.